wire-ios-sync-engine/Source/SessionManager/SessionManager.swift

```diff
@@ -1029,19 +1029,23 @@ public final class SessionManager: NSObject, SessionManagerType {
                     with: coreDataStack
                 )
 
-                triggerSlowSyncIfNeeded(with: userSession)
+                triggerMigrationsNeedsActionsIfNeeded(with: userSession)
 
                 onCompletion(userSession)
             }
         )
     }
 
-    private func triggerSlowSyncIfNeeded(with userSession: ZMUserSession) {
+    /// Executes post migration slow sync or sync resources
+    private func triggerMigrationsNeedsActionsIfNeeded(with userSession: ZMUserSession) {
         let context = userSession.syncContext
         context.perform {
-            if context.readAndResetSlowSyncFlag() {
+            if context.readMigrationNeedsSlowSyncFlag() {
                 userSession.syncStatus.forceSlowSync()
             }
+            if context.readMigrationNeedsSyncResourcesFlag() {
+                userSession.syncStatus.resyncResources()
+            }
```

Comment:
Are these compatible together? Resync might override slow sync. Maybe slow sync takes precedence?

Reply:
you mean like doing a if else if ?